packages/core-contracts/contracts/ownership/Authorizable.sol

```diff
@@ -0,0 +1,32 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./AuthorizableMixin.sol";
+import "./Ownable.sol";
+import "../interfaces/IAuthorizable.sol";
+import "../errors/AddressError.sol";
+import "../errors/ChangeError.sol";
+
+contract Authorizable is Ownable, IAuthorizable, AuthorizableMixin {
+    event AuthorizedChanged(address oldAuthorizad, address newAuthorized);
+
+    function setNewAuthorized(address newAuthorized) public override onlyOwner {
+        AuthorizableStore storage store = _authorizableStore();
+        address oldAuthorized = store.authorized;
+
+        if (newAuthorized == address(0)) {
```

Comment:
This check can be hoisted up in the function to save gas in the case of a revert.